### PR TITLE
galactic: updated ci-job and codecov

### PR DESCRIPTION
**Pull request by JanStaschulat**
- updating the codecov link
- fixing issues regarding the CI workflow

### CODE REVIEW

**Comment by JanStaschulat** [comment]
@pablogs9 , any idea, why the CI job for galactic is not started with the pull request?

**Comment by pablogs9** [comment]
@JanStaschulat I think that the workflows only are available if they are in the `default` branch: https://github.community/t/workflow-files-only-picked-up-from-master/16129

**Comment by christophebedard** [comment]
I think if you change this from `master` to `galactic` (or just remove the line altogether) it should work:

https://github.com/ros2/rclc/blob/de019cf4bfd409ba5488d78ae2c19e1159ae9c83/.github/workflows/ci.yml#L7

**Comment by JanStaschulat** [comment]
but it fails with the following message in the CI workflow:
```

Invoking "sudo rosdep init"
  /usr/bin/sudo rosdep init
  Wrote /etc/ros/rosdep/sources.list.d/20-default.list
  Recommended: please run
  
  	rosdep update
  
Error: Input has invalid distribution names.
```

**Comment by christophebedard** [comment]
> but it fails with the following message in the CI workflow:
> 
> ```
> 
> Invoking "sudo rosdep init"
>   /usr/bin/sudo rosdep init
>   Wrote /etc/ros/rosdep/sources.list.d/20-default.list
>   Recommended: please run
>   
>   	rosdep update
>   
> Error: Input has invalid distribution names.
> ```

yeah, Galactic (release testing) isn't supported by `setup-ros` or `action-ros-ci` just yet. See https://github.com/ros-tooling/setup-ros/pull/387

**Comment by christophebedard** [comment]
@JanStaschulat this should now be supported by `setup-ros` or `action-ros-ci`, although you'll need to use `@master` for now until we create a new release.

**Comment by JanStaschulat** [comment]
> @JanStaschulat this should now be supported by `setup-ros` or `action-ros-ci`, although you'll need to use `@master` for now until we create a new release.

@christophebedard thanks for the information. I upgraded `setup-ros`and `action-ros-ci` to the most recent versions. But it still fails for the `Galactic` branch. I guess, I have to wait until next week when Galactic is officially released?

**Comment by christophebedard** [comment]
If you use `@master`

```yaml
- uses: ros-tooling/setup-ros@master
...
- uses: ros-tooling/action-ros-ci@master
...
```

You can use `galactic` as the `ros_distribution` (with the `use-ros2-testing: true` option for `setup-ros` ).

**Comment by christophebedard** [comment]
..and by upgrading to `@master` you'll need to replace

```yaml
      colcon-mixin-name: coverage-gcc
```

with

```yaml
      ​colcon-defaults​: ​|​
​        {​
​          "build": {​
​            "mixin": ["coverage-gcc"]​
​          }​
​        }​
```

😅

**Comment by JanStaschulat** [comment]
For some reason, the CI jobs don't run.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,19 +25,25 @@ jobs:
...
         package-name: "rclc rclc_examples rclc_lifecycle"
         vcs-repo-file-url: dependencies.repos
         target-ros2-distro: ${{ matrix.ros_distribution }}
-        colcon-mixin-name: coverage-gcc
+        colcon-defaults: |
+        {
+          "build": {
+            "mixin": [ "coverage-gcc" ]
+          }
+        }
```
Comment (christophebedard) [inline .github/workflows/ci.yml:46]: the indentation is wrong here. The 5 lines above should be indented by two spaces more.

**Review by pablogs9 — APPROVED** [review]

**Comment by christophebedard** [comment]
Galactic has been released and the actions have just been updated to fully support it, so you can now remove `use-ros2-testing: true` and use `ros-tooling/setup-ros@0.2.0` (or `@v0.2`) and `ros-tooling/action-ros-ci@0.2.1` (or `@v0.2`).

**Comment by JanStaschulat** [comment]
@christophebedard Thanks for the notification. I updated the ci-job. Let's see if the bloom release works out. I updated also the branch version for rclc bloom-release in rosdistro: https://github.com/ros/rosdistro/pull/29683

**Comment by JanStaschulat** [comment]
@christophebedard Do you have an idea, why the bloom realease `Gpr__rclc__ubuntu_focal_amd64` is not running?

**Comment by christophebedard** [comment]
> @christophebedard Do you have an idea, why the bloom realease for Galactic `Gpr__rclc__ubuntu_focal_amd64` is not running?

I don't know, sorry. Looks like it's running for other/newer PRs though.

**Comment by JanStaschulat** [comment]
@christophebedard i merged it without the test on the build farm. I created a new bloom release (see issue https://github.com/ros2/rclc/issues/96) for galactic.

**Pull request merged**